Inocompaibility with Gradle 8

While trying to upgrade https://github.com/grails/grails-gsp to Gradle 8, the following error occurs: 

```
Caused by: groovy.lang.MissingPropertyException: Could not set unknown property 'classifier' for task ':grails-plugin-gsp:javadocJar' of type org.gradle.api.tasks.bundling.Jar.
	at org.gradle.internal.metaobject.AbstractDynamicObject.setMissingProperty(AbstractDynamicObject.java:118)
	at org.gradle.internal.metaobject.ConfigureDelegate.setProperty(ConfigureDelegate.java:102)
	at org.gradle.internal.classpath.declarations.GroovyDynamicDispatchInterceptors.intercept_setGroovyObjectProperty(GroovyDynamicDispatchInterceptors.java:57)
	at org.grails.gradle.plugin.core.GrailsPluginGradlePlugin$_configureAstSources_closure7.doCall$original(GrailsPluginGradlePlugin.groovy:177)
	at org.grails.gradle.plugin.core.GrailsPluginGradlePlugin$_configureAstSources_closure7.doCall(GrailsPluginGradlePlugin.groovy)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at org.gradle.util.internal.ClosureBackedAction.execute(ClosureBackedAction.java:73)
	at org.gradle.util.internal.ConfigureUtil.configureTarget(ConfigureUtil.java:155)
	at org.gradle.util.internal.ConfigureUtil.configureSelf(ConfigureUtil.java:131)
	at org.gradle.api.internal.AbstractTask.configure(AbstractTask.java:702)
	at org.gradle.api.DefaultTask.configure(DefaultTask.java:319)
	at org.gradle.api.Task$configure.call(Unknown Source)
	at org.grails.gradle.plugin.core.GrailsPluginGradlePlugin.configureAstSources(GrailsPluginGradlePlugin.groovy:176)
	at org.grails.gradle.plugin.core.GrailsPluginGradlePlugin.apply(GrailsPluginGradlePlugin.groovy:64)
	at org.grails.gradle.plugin.core.GrailsPluginGradlePlugin.apply(GrailsPluginGradlePlugin.groovy)
	at org.gradle.api.internal.plugins.ImperativeOnlyPluginTarget.applyImperative(ImperativeOnlyPluginTarget.java:43)
	at org.gradle.api.internal.plugins.RuleBasedPluginTarget.applyImperative(RuleBasedPluginTarget.java:51)
	at org.gradle.api.internal.plugins.DefaultPluginManager.addPlugin(DefaultPluginManager.java:190)
	at org.gradle.api.internal.plugins.DefaultPluginManager.access$100(DefaultPluginManager.java:54)
	at org.gradle.api.internal.plugins.DefaultPluginManager$AddPluginBuildOperation.run(DefaultPluginManager.java:285)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:29)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$1.execute(DefaultBuildOperationRunner.java:26)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:66)
	at org.gradle.internal.operations.DefaultBuildOperationRunner$2.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:157)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.execute(DefaultBuildOperationRunner.java:59)
	at org.gradle.internal.operations.DefaultBuildOperationRunner.run(DefaultBuildOperationRunner.java:47)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:68)
	at org.gradle.api.internal.plugins.DefaultPluginManager.lambda$doApply$0(DefaultPluginManager.java:170)
	at org.gradle.internal.code.DefaultUserCodeApplicationContext.apply(DefaultUserCodeApplicationContext.java:43)
	at org.gradle.api.internal.plugins.DefaultPluginManager.doApply(DefaultPluginManager.java:169)
	... 246 more


Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

For more on this, please refer to https://docs.gradle.org/8.6/userguide/command_line_interface.html#sec:command_line_warnings in the Gradle documentation.

```

--- a/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/core/GrailsGradlePlugin.groovy
@@ -659,10 +659,10 @@ class GrailsGradlePlugin extends GroovyPlugin {
                     pathingJar = createPathingJarTask(project, "pathingJar", runtime)
                 }
 
-                FileCollection pathingClasspath = project.files("${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJar.archivePath) + mainFiles
+                FileCollection pathingClasspath = project.files("${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJar.archiveFile) + mainFiles
 
                 Jar pathingJarCommand = createPathingJarTask(project, "pathingJarCommand", runtime, console)
-                FileCollection pathingClasspathCommand = project.files("${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJarCommand.archivePath) + mainFiles
+                FileCollection pathingClasspathCommand = project.files("${project.buildDir}/resources/main", "${project.projectDir}/gsp-classes", pathingJarCommand.archiveFile) + mainFiles
 
                 GrailsExtension grailsExt = project.extensions.getByType(GrailsExtension)
 

--- a/src/main/groovy/org/grails/gradle/plugin/core/GrailsPluginGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/core/GrailsPluginGradlePlugin.groovy
@@ -120,7 +120,7 @@ class GrailsPluginGradlePlugin extends GrailsGradlePlugin {
         def taskContainer = project.tasks
         if(taskContainer.findByName('sourcesJar') == null) {
             def jarTask = taskContainer.create("sourcesJar", Jar)
-            jarTask.classifier = 'sources'
+            jarTask.archiveClassifier.set('sources')
             jarTask.from SourceSets.findMainSourceSet(project).allSource
         }
     }
@@ -174,7 +174,7 @@ class GrailsPluginGradlePlugin extends GrailsGradlePlugin {
         if (groovydocTask) {
             if( taskContainer.findByName('javadocJar') == null) {
                 taskContainer.create("javadocJar", Jar).configure {
-                    classifier = 'javadoc'
+                    archiveClassifier.set('javadoc')
                     from groovydocTask.outputs
                 }.dependsOn(javadocTask)
             }

--- a/src/main/groovy/org/grails/gradle/plugin/doc/PublishGuideTask.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/doc/PublishGuideTask.groovy
@@ -22,7 +22,6 @@ import org.gradle.api.tasks.InputFile
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.compile.AbstractCompile
-import org.gradle.work.InputChanges
 
 /**
  * A task used to publish the user guide if a publin that is in GDoc format
@@ -67,7 +66,7 @@ class PublishGuideTask extends AbstractCompile {
 
     @CompileDynamic
     @TaskAction
-    void compile(InputChanges inputChanges) {
+    void compile() {
         def urls = getClasspath().files.collect() { File f -> f.toURI().toURL() }
 
         URLClassLoader classLoader = new URLClassLoader(urls as URL[], (ClassLoader) null)

--- a/src/main/groovy/org/grails/gradle/plugin/profiles/GrailsProfileGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/profiles/GrailsProfileGradlePlugin.groovy
@@ -151,7 +151,7 @@ class GrailsProfileGradlePlugin implements Plugin<Project> {
                 spec.into("skeleton")
             }
             jar.archiveClassifier.set("sources")
-            jar.destinationDir = new File(project.buildDir, "libs")
+            jar.destinationDirectory.set(new File(project.buildDir, "libs"))
             jar.setDescription("Assembles a jar archive containing the profile sources.")
             jar.setGroup(BUILD_GROUP)
 

--- a/src/main/groovy/org/grails/gradle/plugin/profiles/GrailsProfileGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/profiles/GrailsProfileGradlePlugin.groovy
@@ -150,7 +150,7 @@ class GrailsProfileGradlePlugin implements Plugin<Project> {
             jar.from(skeletonsDir) { CopySpec spec ->
                 spec.into("skeleton")
             }
-            jar.classifier = "sources"
+            jar.archiveClassifier.set("sources")
             jar.destinationDir = new File(project.buildDir, "libs")
             jar.setDescription("Assembles a jar archive containing the profile sources.")
             jar.setGroup(BUILD_GROUP)

--- a/src/main/groovy/org/grails/gradle/plugin/profiles/internal/GrailsProfilePublishGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/profiles/internal/GrailsProfilePublishGradlePlugin.groovy
@@ -49,7 +49,7 @@ class GrailsProfilePublishGradlePlugin extends GrailsCentralPublishGradlePlugin 
         project.tasks.create("javadocJar", Jar, (Action) { Jar jar ->
             jar.from(tempReadmeForJavadoc)
             jar.archiveClassifier.set("javadoc")
-            jar.destinationDir = new File(project.buildDir, "libs")
+            jar.destinationDirectory.set(new File(project.buildDir, "libs"))
             jar.setDescription("Assembles a jar archive containing the profile javadoc.")
             jar.setGroup(BUILD_GROUP)
         })

--- a/src/main/groovy/org/grails/gradle/plugin/profiles/internal/GrailsProfilePublishGradlePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/profiles/internal/GrailsProfilePublishGradlePlugin.groovy
@@ -48,7 +48,7 @@ class GrailsProfilePublishGradlePlugin extends GrailsCentralPublishGradlePlugin 
         tempReadmeForJavadoc << "https://central.sonatype.org/publish/requirements/#supply-javadoc-and-sources"
         project.tasks.create("javadocJar", Jar, (Action) { Jar jar ->
             jar.from(tempReadmeForJavadoc)
-            jar.classifier = "javadoc"
+            jar.archiveClassifier.set("javadoc")
             jar.destinationDir = new File(project.buildDir, "libs")
             jar.setDescription("Assembles a jar archive containing the profile javadoc.")
             jar.setGroup(BUILD_GROUP)

--- a/src/main/groovy/org/grails/gradle/plugin/profiles/tasks/ProfileCompilerTask.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/profiles/tasks/ProfileCompilerTask.groovy
@@ -26,7 +26,6 @@ import org.gradle.api.file.FileTree
 import org.gradle.api.file.FileVisitDetails
 import org.gradle.api.tasks.*
 import org.gradle.api.tasks.compile.AbstractCompile
-import org.gradle.work.InputChanges
 import org.grails.cli.profile.commands.script.GroovyScriptCommand
 import org.grails.cli.profile.commands.script.GroovyScriptCommandTransform
 import org.grails.gradle.plugin.profiles.GrailsProfileGradlePlugin
@@ -79,7 +78,7 @@ class ProfileCompilerTask extends AbstractCompile {
     }
 
     @TaskAction
-    void execute(InputChanges inputChanges) {
+    void execute() {
 
         boolean profileYmlExists = config?.exists()
 

--- a/src/main/groovy/org/grails/gradle/plugin/util/SourceSets.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/util/SourceSets.groovy
@@ -2,7 +2,7 @@ package org.grails.gradle.plugin.util
 
 import groovy.transform.CompileStatic
 import org.gradle.api.Project
-import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.plugins.JavaPluginExtension
 import org.gradle.api.tasks.SourceSet
 import org.gradle.api.tasks.SourceSetContainer
 
@@ -35,7 +35,7 @@ class SourceSets {
     }
 
     static SourceSetContainer findSourceSets(Project project) {
-        JavaPluginConvention plugin = project.getConvention().getPlugin(JavaPluginConvention)
+        JavaPluginExtension plugin = project.extensions.getByType(JavaPluginExtension)
         SourceSetContainer sourceSets = plugin?.sourceSets
         return sourceSets
     }

--- a/src/main/groovy/org/grails/gradle/plugin/web/gsp/GroovyPageCompileTask.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/web/gsp/GroovyPageCompileTask.groovy
@@ -9,7 +9,6 @@ import org.gradle.api.tasks.InputDirectory
 import org.gradle.api.tasks.Optional
 import org.gradle.api.tasks.TaskAction
 import org.gradle.api.tasks.compile.AbstractCompile
-import org.gradle.work.InputChanges
 
 /**
  * A task for compiling GSPs
@@ -46,7 +45,7 @@ class GroovyPageCompileTask extends AbstractCompile {
 
     @TaskAction
     @CompileDynamic
-    protected void execute(InputChanges inputChanges) {
+    protected void execute() {
 
         def compileTask = this
         Project gradleProject = project

--- a/src/main/groovy/org/grails/gradle/plugin/web/gsp/GroovyPagePlugin.groovy
+++ b/src/main/groovy/org/grails/gradle/plugin/web/gsp/GroovyPagePlugin.groovy
@@ -73,7 +73,7 @@ class GroovyPagePlugin implements Plugin<Project> {
             GrailsExtension grailsExt = project.extensions.getByType(GrailsExtension)
             if (grailsExt.pathingJar && Os.isFamily(Os.FAMILY_WINDOWS)) {
                 Jar pathingJar = (Jar) allTasks.findByName('pathingJar')
-                allClasspath = project.files("${project.buildDir}/classes/groovy/main", "${project.buildDir}/resources/main", pathingJar.archivePath)
+                allClasspath = project.files("${project.buildDir}/classes/groovy/main", "${project.buildDir}/resources/main", pathingJar.archiveFile)
                 compileGroovyPages.dependsOn(pathingJar)
                 compileGroovyPages.setClasspath(allClasspath)
                 compileWebappGroovyPages.dependsOn(pathingJar)

